src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1454,6 +1454,14 @@ bool MatchesConstructorParameter(ParameterGenerationSpec paramSpec)
             private TypeRef? GetConverterTypeFromJsonConverterAttribute(INamedTypeSymbol contextType, ISymbol declaringSymbol, AttributeData attributeData)
             {
                 Debug.Assert(_knownSymbols.JsonConverterAttributeType.IsAssignableFrom(attributeData.AttributeClass));
+
+                if (!SymbolEqualityComparer.Default.Equals(attributeData.AttributeClass, _knownSymbols.JsonConverterAttributeType))
+                {
+                    ReportDiagnostic(DiagnosticDescriptors.DerivedJsonConverterAttributesNotSupported, attributeData.GetDiagnosticLocation());
```

Comment:
>attributeData.GetDiagnosticLocation() [](http://example.com/codeflow?start=103&length=37)

I recall you have issue tracking if we can have the location outside the compilation. right? just checking.

Reply:
Yes, https://github.com/dotnet/runtime/issues/88424